lib/internal/test_runner/mock/mock_timers.js

```diff
@@ -146,7 +146,7 @@ class MockTimers {
       // ECMA-262 Edition 5.1 section 15.9.2.
       // and ECMA-262 Edition 14 Section 21.4.2.1
       // this replaces 'this instanceof MockDate' with a more reliable check
-      if (isValidDateWithGetTime(this)) {
+      if (!this) {
```

Comment:
I added this validation which appears to be working in most of the cases, however I'm not sure if it works in all the possible edge cases. 

For some reason the check I was using didn't work, I was also trying to see why, apparently the validation with `isValidDateWithGetTime` was crashing with a V8 error before it could return, after many attempts to fix it with different approaches, the one that proved to be more reliable to check if the function is not called as an instance was this one.

Reply:
This is _almost_ right but fails for `.call`. A better check would probably be for `new.target`.

```suggestion
      if (!new.target) {
```

Reply:
Thanks for the reply! I didn't know this possible option! Let's see if all the tests pass :)